tests/backend/unit/test_project_structure.py

```
@@ -0,0 +1,702 @@
+                cycle_descriptions.append(cycle_str)
+            
+            pytest.fail(
+                f"Circular dependencies detected:\n" + 
+                "\n".join(cycle_descriptions)
+            )
+
+
+class TestProjectBestPractices:
+    """Test suite for Python and JavaScript/TypeScript best practices."""
+    
+    def test_no_hardcoded_secrets(self, project_root: Path):
```

Ruff (0.11.9)

`test_no_hardcoded_secrets` is too complex (12 > 10)

(C901)